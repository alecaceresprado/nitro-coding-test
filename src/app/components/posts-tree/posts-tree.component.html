<div class="posts-list">
  <div class="posts-list__title">Posts</div>
  <div
    class="posts-list__parent-level"
    *ngFor="let parent of parentLevel"
    [class.posts-list__parent-level--selected]="parent === selectedParent"
    (click)="handleParentClick(parent)"
  >
    {{ parent }}
    <div *ngIf="parent === selectedParent">
      <div
        *ngFor="let post of postsToList"
        class="post-text"
        [class.post-text--selected]="post.id === selectedPostId"
        (click)="handlePostClick.emit(post.id)"
      >
        {{ post.text }}
      </div>
    </div>
  </div>
</div>
