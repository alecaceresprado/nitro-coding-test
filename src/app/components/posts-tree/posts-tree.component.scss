@import "../../../styles/vars";

.posts-list {
  &__title {
    @include sectionTitle();
  }
  &__parent-level {
    margin: 0 1rem;
    padding: 0 1rem;

    &--selected {
      background: lightblue;
      color: white;
    }
  }
}
.post-text {
  padding: .5rem 1rem 0;
  text-overflow: ellipsis;
  white-space: nowrap;
  overflow: hidden;

  &:last-child {
    padding-bottom: .5rem;
  }

  &--selected::before {
    content: "Â»";
    margin-left: -.75rem;
  }
}
