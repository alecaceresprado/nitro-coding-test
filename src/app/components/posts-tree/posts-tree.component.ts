import { Component, OnChanges, Input, Output, EventEmitter } from '@angular/core';

import { PostTree, Post } from '@models';

@Component({
  selector: 'app-posts-tree',
  templateUrl: './posts-tree.component.html',
  styleUrls: ['./posts-tree.component.scss'],
})
export class PostsTreeComponent implements OnChanges {
  @Input() public postTree: PostTree;
  @Input() public selectedPostId: number;
  @Output() public handlePostClick: EventEmitter<number> = new EventEmitter();

  public parentLevel: string[];
  public selectedParent: string;
  public postsToList: Post[];

  constructor() {}

  ngOnChanges(): void {
    this.parentLevel = Object.keys(this.postTree);
  }

  public handleParentClick(parent: string): void {
    this.selectedParent = parent;
    this.postsToList = this.postTree[parent];
  }
}
