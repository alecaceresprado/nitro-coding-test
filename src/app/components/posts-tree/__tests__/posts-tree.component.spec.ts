import { async, ComponentFixture, TestBed } from '@angular/core/testing';

import { PostsTreeComponent } from '../posts-tree.component';

const postTreeMock = {
  'Week 4': [
    {
      id: 5,
      location: 'Dublin',
      time: '1553080742',
      author: 'Happy Manager',
      text:
        'A modern PDF annotator that canaccommodate all of the cooks in a very busy kitchen is what your employees really need.',
    },
    {
      id: 6,
      location: 'Dublin',
      time: '1553099742',
      author: 'Happy Manager',
      text:
        'An integratedproductivity solution breaks information through barriers and allows workers to collaborate in real time.',
    },
  ],
  'Week 24': [
    {
      id: 1,
      location: 'San Francisco',
      time: '1591699872486',
      author: 'Happy User',
      text:
        'Proper PDF conversion ensures that every element of your document remains just as you left it.',
    },
    {
      id: 4,
      location: 'Sydney',
      time: '1591699872999',
      author: 'Happy Developer',
      text:
        'An expectation of digital efficiency has become the norm in our daily lives',
    },
  ],
  'Week 31': [
    {
      id: 2,
      location: 'San Francisco',
      time: '1501690872486',
      author: 'Happy User',
      text:
        'The modernworkplace is increasingly digital, and workflows are constantly evolving. ',
    },
    {
      id: 3,
      location: 'SanFrancisco',
      time: '1501690872999',
      author: 'Happy Developer',
      text: 'Digital transformation isnâ€™t justa buzzword',
    },
  ],
};

describe('PostsTreeComponent', () => {
  let component: PostsTreeComponent;
  let fixture: ComponentFixture<PostsTreeComponent>;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      declarations: [PostsTreeComponent],
    }).compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(PostsTreeComponent);
    component = fixture.componentInstance;
    component.postTree = postTreeMock;
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('should render', () => {
    expect(fixture).toMatchSnapshot();
  });

  it('should handle parent click', () => {
    component.handleParentClick('Week 4');
    fixture.detectChanges();
    expect(component.selectedParent).toEqual('Week 4');
    expect(component.postsToList).toEqual(postTreeMock['Week 4']);
  });
});
