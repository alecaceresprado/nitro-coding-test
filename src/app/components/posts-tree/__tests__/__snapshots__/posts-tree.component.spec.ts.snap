// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostsTreeComponent should render 1`] = `
<app-posts-tree
  handlePostClick={[Function EventEmitter]}
  postTree={[Function Object]}
>
  <div
    class="posts-list"
  >
    <div
      class="posts-list__title"
    >
      Posts
    </div>
    
  </div>
</app-posts-tree>
`;
